### PR TITLE
Add handling for 401, 403, 408, and 500 status from charon

### PR DESCRIPTION
**Pull request by ilaflamme-va**

### CODE REVIEW

File: vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/controller/WebExceptionHandlerTest.java [hunk 1]
```diff
@@ -53,6 +57,56 @@ void badRequest() {
...
 
+  @Test
+  void forbidden() {
+    HttpClientErrorException forbidden =
+        HttpClientErrorException.Forbidden.create(HttpStatus.FORBIDDEN, null, null, null, null);
+    OperationOutcome outcome =
+        new WebExceptionHandler("").handleForbidden(forbidden, mock(HttpServletRequest.class));
+    assertThat(outcome.id(null).extension(null))
+        .isEqualTo(
+            OperationOutcome.builder()
```
Comment (bschofield-va) [inline vista-fhir-query/src/test/java/gov/va/api/health/vistafhirquery/service/controller/WebExceptionHandlerTest.java:68]: These OOs seem to differ in _code_ only, perhaps avoid the duplication with an `operationOutcome("forbidden")` helper

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/WebExceptionHandler.java [hunk 2]
```diff
@@ -207,6 +218,12 @@ public OperationOutcome handleSnafu(Exception e, HttpServletRequest request) {
     return responseFor("exception", e, request, emptyList(), true);
   }
 
+  @ExceptionHandler({HttpClientErrorException.Unauthorized.class})
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/WebExceptionHandler.java:221]: I think we wanted some sort of javadoc/ comment explained why we're mapping 401 to 500 right?

Comment (jhulbert-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/WebExceptionHandler.java:221]: I see it in forbidden, should we just group those together with the log?

Comment (ilaflamme-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/WebExceptionHandler.java:221]: The log statement works, ya?

Comment (jhulbert-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/WebExceptionHandler.java:221]: I mean should it move to the ISE section and use the same log statement forbidden does?

**Review by jhulbert-va — APPROVED** [review]

**Pull request merged**